kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriterTest.java

```diff
@@ -0,0 +1,79 @@
+package com.wepay.kafka.connect.bigquery.write.row;
+
+/*
+ * Copyright 2016 WePay, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import com.google.cloud.bigquery.BigQueryError;
+import com.google.cloud.bigquery.InsertAllRequest;
+import com.wepay.kafka.connect.bigquery.exception.BigQueryConnectException;
+import com.wepay.kafka.connect.bigquery.utils.PartitionedTableId;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.*;
+
+public class BigQueryWriterTest {
+    private Logger logger;
+    private PartitionedTableId testTableId ;
+    private List<InsertAllRequest.RowToInsert> testRows;
+    private String testTopic;
+
+    @Before
+    public void init() {
+        logger = LoggerFactory.getLogger(BigQueryWriterTest.class);
+        testTableId = new PartitionedTableId.Builder("testDataset", "testBaseTable").build();
+        testRows = new ArrayList<>();
+        testRows.add(mock(InsertAllRequest.RowToInsert.class));
+        testRows.add(mock(InsertAllRequest.RowToInsert.class));
+        testTopic = "testTopic";
+    }
+
+    @Test(expected = BigQueryConnectException.class)
+    public void testWriteRowsShouldThrowExcepiton() throws InterruptedException {
```

Comment:
write rows should throw exception if...? Can you elaborate in a comment a bit more detail on what this is testing? (also excepiton -> exception)

Reply:
I want to test this 'if ... else if ... else' block here: https://github.com/wepay/kafka-connect-bigquery/pull/108/commits/3d51fb6e6edf7b7811c3ca785437c410ab998bec#diff-f00fa7ac2c3aa6ac09bf9381986fc494R118
```
if succeed, return;
else failed
    if partial fail, update rows, (will throw BigQueryConnectionException once retry == 0)
    else throw BigQueryConnectionException
```
Currently, I only covers "succeed" and 'failed' cases, though it would be ideal to cover 3 cases. I am not sure how to discern last 2 case in unit test, any suggestion?

Reply:
👍 great idea!